File: Generating/factory/cpp/EthereumMarket.cpp
#ifndef ethereum_instance_hpp
#define ethereum_instance_hpp

#include "CoinInstance.hpp"
#include "MarketFactory.cpp"
#include <string>

using namespace std;

class Ethereum : public CoinInstance
{
  public:
    Ethereum(void) {
      this->name = "Ethereum";
      this->network = "ropsten";
      this->balance = 0;
    };
    ~Ethereum(void) {};
    
    string getName() {
      return this->name;
    };

    int getBalance() {
      return this->balance;
    };

    string getNetwork() {
      return this->network;
    };

    void setBalance(int new_balance) {
      this->balance = new_balance;
    };
};

class EthereumMarket : public MarketFactory
{
  public:
    CoinInstance * getCoinInstance() {
      return new Ethereum();
    }
};

#endif

File: Generating/factory/rust/ethereum_market.rs
use traits::*;
use configuration::*;

#[derive(Debug)]
pub struct Ethereum {
  name: String,
  balance: i32,
  network: String
}

pub trait EthereumFactory {
  fn new(config: &EthereumConfig) -> Self;
}

impl EthereumFactory for Ethereum {
  fn new(config: &EthereumConfig) -> Self {
    Ethereum {
      name: "Ethereum".to_string(),
      balance: 0,
      network: config.get_network()
    }
  }
}

impl CoinInstance for Ethereum {
  fn get_name(&self) -> String {
    self.name.clone()
  }

  fn get_balance(&self) -> i32 {
    self.balance
  }

  fn get_network(&self) -> String {
    self.network.clone()
  }

  fn set_balance(&mut self, new_balance: i32) {
    self.balance = new_balance;
  }
}